### ISSUE TITLE
v0.8.4 dual stack static pod fails with runtime error

### ISSUE THREAD

**Issue by work-smalin** [kube-vip/kube-vip#971]
**Describe the bug**
kube-vip v0.8.4 dual stack static pod fails to run on any control plane.
After the IP addresses are activated a runtime error occurs.

**To Reproduce**
Generate a manifest, place it in /etc/kubernetes/manifests, change to super-admin.conf, initialize the cluster and change back to admin.conf, eg:
```bash
export INTERFACE=enp1s0
KVVERSION=v0.8.4
KUBEVIP="docker run --network host --rm ghcr.io/kube-vip/kube-vip:$KVVERSION"
export VIP=10.10.10.12,fd00:4269:6943::12

$KUBEVIP manifest pod \
    --interface $INTERFACE \
    --address $VIP \
    --controlplane \
    --arp \
    --leaseDuration 15 \
    --leaseRenewDuration 10 \
    --leaseRetry 2 \
    --leaderElection > kube-vip.yaml

scp kube-vip.yaml root@host:/etc/kubernetes/manifests
ssh root@host
sed -i 's#path: /etc/kubernetes/admin.conf#path: /etc/kubernetes/super-admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml
kubeadm init --control-plane-endpoint ...
sed -i 's#path: /etc/kubernetes/super-admin.conf#path: /etc/kubernetes/admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml
```

**Expected behavior**
kube-vip v0.8.4 runs dual stack configuration without errors.

**Environment (please complete the following information):**
 - OS/Distro: Debian 12.7
 - Kubernetes Version: 1.31.1
 - Kube-vip Version: 0.8.4

**Screenshots**
Full log of kube-vip pod:
```
time="2024-10-23T06:27:57Z" level=info msg="Starting kube-vip.io [v0.8.4]"
time="2024-10-23T06:27:57Z" level=info msg="namespace [kube-system], Mode: [ARP], Features(s): Control Plane:[true], Services:[false]"
time="2024-10-23T06:27:57Z" level=info msg="prometheus HTTP server started"
time="2024-10-23T06:27:57Z" level=info msg="Using node name [cp1]"
time="2024-10-23T06:27:57Z" level=info msg="Starting Kube-vip Manager with the ARP engine"
time="2024-10-23T06:27:57Z" level=info msg="Beginning cluster membership, namespace [kube-system], lock name [plndr-cp-lock], id [cp1]"
I1021 06:27:57.016794       1 leaderelection.go:250] attempting to acquire leader lease kube-system/plndr-cp-lock...
I1021 06:27:57.030441       1 leaderelection.go:260] successfully acquired lease kube-system/plndr-cp-lock
time="2024-10-23T06:27:57Z" level=info msg="Node [cp1] is assuming leadership of the cluster"
time="2024-10-23T06:27:57Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [10.10.10.12/enp1s0]"
time="2024-10-23T06:27:57Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [fd00:4269:6943::12/enp1s0]"
time="2024-10-23T06:27:57Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
time="2024-10-23T06:27:57Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc000446f80]}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1783770]

goroutine 45 [running]:
github.com/kube-vip/kube-vip/pkg/vip.(*NdpResponder).SendGratuitous(0x0, {0xc000056fc0, 0x16})
        /src/pkg/vip/ndp.go:53 +0x110
github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).ensureIPAndSendGratuitous(0xc000418240, {0xc000184340, 0x6}, 0x0)
        /src/pkg/cluster/service.go:421 +0x3e5
github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).vipService.func3({0x2332fc0, 0xc000360140}, 0x0)
        /src/pkg/cluster/service.go:133 +0x26a
created by github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).vipService in goroutine 49
        /src/pkg/cluster/service.go:112 +0x9eb
```

**`Kube-vip.yaml`:**
```yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  name: kube-vip
  namespace: kube-system
spec:
  containers:
  - args:
    - manager
    env:
    - name: vip_arp
      value: "true"
    - name: port
      value: "6443"
    - name: vip_nodename
      valueFrom:
        fieldRef:
          fieldPath: spec.nodeName
    - name: vip_interface
      value: enp1s0
    - name: vip_cidr
      value: 32,128
    - name: dns_mode
      value: first
    - name: cp_enable
      value: "true"
    - name: cp_namespace
      value: kube-system
    - name: vip_leaderelection
      value: "true"
    - name: vip_leasename
      value: plndr-cp-lock
    - name: vip_leaseduration
      value: "15"
    - name: vip_renewdeadline
      value: "10"
    - name: vip_retryperiod
      value: "2"
    - name: address
      value: 10.10.10.12,fd00:4269:6943::12
    - name: prometheus_server
      value: :2112
    image: ghcr.io/kube-vip/kube-vip:v0.8.4
    imagePullPolicy: IfNotPresent
    name: kube-vip
    resources: {}
    securityContext:
      capabilities:
        add:
        - NET_ADMIN
        - NET_RAW
    volumeMounts:
    - mountPath: /etc/kubernetes/admin.conf
      name: kubeconfig
  hostAliases:
  - hostnames:
    - kubernetes
    ip: 127.0.0.1
  hostNetwork: true
  volumes:
  - hostPath:
      path: /etc/kubernetes/admin.conf
    name: kubeconfig
status: {}
```

**Additional context**
- Usage of super-admin.conf due to the kubernetes version and only on first control plane  (see issue #684).
- When reproducing the manifest with kube-vip v0.8.3 the pod runs flawlessly on every control plane.

BTW: thanks for the project and the great work :)

**Comment by p-strusiewiczsurmacki-mobica** [comment]
Hi @work-smalin ,

Sorry, it seems I've messed NDP up in one of previous commits. I've just created PR with a fix, however if you are be able to build the image and test it on your setup it would be highly appreciated!

**Comment by work-smalin** [comment]
Hi @p-strusiewiczsurmacki-mobica,

thanks for your quick reply :)
Yes, after building the image with your fix, kube-vip v0.8.4 seems to work as expected:
```
2024-10-23T19:04:55.896457020+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Starting kube-vip.io [v0.8.4]"
2024-10-23T19:04:55.896600973+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="namespace [kube-system], Mode: [ARP], Features(s): Control Plane:[true], Services:[false]"
2024-10-23T19:04:55.896710406+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="prometheus HTTP server started"
2024-10-23T19:04:55.897285114+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Using node name [cp1]"
2024-10-23T19:04:55.899827903+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Starting Kube-vip Manager with the ARP engine"
2024-10-23T19:04:55.900924748+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Beginning cluster membership, namespace [kube-system], lock name [plndr-cp-lock], id [cp1]"
2024-10-23T19:04:55.901165502+02:00 stderr F I1023 17:04:55.901138       1 leaderelection.go:250] attempting to acquire leader lease kube-system/plndr-cp-lock...
2024-10-23T19:04:55.918823819+02:00 stderr F I1023 17:04:55.918779       1 leaderelection.go:260] successfully acquired lease kube-system/plndr-cp-lock
2024-10-23T19:04:55.918955443+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Node [cp1] is assuming leadership of the cluster"
2024-10-23T19:04:55.919544057+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [10.10.10.12/enp1s0]"
2024-10-23T19:04:55.919917514+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [fd00:4269:6943::12/enp1s0]"
2024-10-23T19:04:55.920063352+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
2024-10-23T19:04:55.920302983+02:00 stderr F time="2024-10-23T17:04:55Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc0000502e0]}"
2024-10-23T19:04:58.920877771+02:00 stderr F time="2024-10-23T17:04:58Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
2024-10-23T19:04:58.920877771+02:00 stderr F time="2024-10-23T17:04:58Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc00013e7e0]}"
2024-10-23T19:05:01.921343452+02:00 stderr F time="2024-10-23T17:05:01Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
2024-10-23T19:05:01.921343452+02:00 stderr F time="2024-10-23T17:05:01Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc000050900]}"
2024-10-23T19:05:04.922525458+02:00 stderr F time="2024-10-23T17:05:04Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
2024-10-23T19:05:04.922525458+02:00 stderr F time="2024-10-23T17:05:04Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc0002beba0]}"
...
```
